internal/cephfs/core/volume.go

```diff
@@ -201,19 +212,22 @@ const (
 type localClusterState struct {
 	// set the enum value i.e., unknown, supported,
 	// unsupported as per the state of the cluster.
-	resizeState                 operationState
-	subVolMetadataState         operationState
-	subVolSnapshotMetadataState operationState
+	resizeState                 operationStateMutex
+	subVolMetadataState         operationStateMutex
+	subVolSnapshotMetadataState operationStateMutex
```

Comment:
this may be overdoing it, is the `subVolumeGroupsRWMutex` of this struct not sufficient to protect its members?

Reply:
> this may be overdoing it, is the `subVolumeGroupsRWMutex` of this struct not sufficient to protect its members?

No, all are independent parameters.
+ with R-Locks it should basically not have much overhead.

Reply:
It is not about the performance overhead. I am worried about the complexity that looks unneeded.